addon/utils/validators/shape.js

```diff
@@ -8,14 +8,15 @@ const {get, typeOf} = Ember
 import logger from '../logger'
 
 export default function (validators, ctx, name, value, def, logErrors, throwErrors) {
+  const shape = JSON.stringify(value, null, ' ')
```

Comment:
Yes, recursive references, which happens if you have a DOM element anywhere. I seem to recall having issues with properties that are functions too, but that may have been a while ago.